### PR TITLE
bugfix: ribbon menu translations hot update

### PR DESCRIPTION
**Pull request by shun2wang**
Fix: https://github.com/jasp-stats/INTERNAL-jasp/issues/2372

So we just remove/renew the menu ribbon button to "reload" it to apply the translations.

### CODE REVIEW

**Comment by boutinb** [comment]
Really nice!
Just remove the 6 Modules/pkgdepends.lock of your PR.

**Comment by shun2wang** [comment]
>Just remove the 6 Modules/pkgdepends.lock of your PR.

Well, it should be updated all because it block build on Linux.

Or someone can update all packages in `pkgdepends.lock`.

**Comment by RensDofferhoff** [comment]
@shun2wang current pkgdepends does not build for you?

**Comment by shun2wang** [comment]
yeah, you can also find it from any failed actions: https://github.com/jasp-stats/jasp-desktop/actions/runs/10598393408

old packages has build problems with gcc  version maybe.

**Comment by RensDofferhoff** [comment]
Interesting! I have not seen these issues on my machines

@boutinb  The updates are alright

File: Desktop/modules/ribbonmodel.cpp [hunk 1]
```diff
@@ -110,6 +111,27 @@ void RibbonModel::loadModules(std::vector<std::string> commonModulesToLoad, std:
...
+
+    for (const auto& button : allButtons)
+    {
+        removeRibbonButtonModel(button);
+    }
+
+    loadModules();
+}
+
 void RibbonModel::addRibbonButtonModelFromDynamicModule(Modules::DynamicModule * module)
```
Comment (boutinb) [inline Desktop/modules/ribbonmodel.cpp:135]: I don't think this is the right way to do this. The loadModules(I) will recreate the modules, make some memory leakage (_entriesInsert, _entriesDelete, etc...), this is quite dangerous I think.
I don't understand why the 'special ribbon entries' are not translated, and the module ribbon buttons are. I remember that @JorisGoosen knew why. Maybe it is better to wait for his return of his holidays.

**Comment by JorisGoosen** [comment]
Thanks for making this @shun2wang but Ive made another PR for this at https://github.com/jasp-stats/jasp-desktop/pull/5653/
That one obviates the need for destroying and recreating the buttons. 
And also doesnt lead to possible problems with (re)loading the modules etc. Which could lead to problems with development modules, it also feels a bit more straightforward this way.

**Pull request closed without merging**